x-pack/plugins/alerting/server/routes/bulk_delete_rules.test.ts

```diff
@@ -0,0 +1,129 @@
+
+import { bulkDeleteRulesRoute } from './bulk_delete_rules';
+import { licenseStateMock } from '../lib/license_state.mock';
+import { mockHandlerArguments } from './_mock_handler_arguments';
+import { rulesClientMock } from '../rules_client.mock';
+import { RuleTypeDisabledError } from '../lib/errors/rule_type_disabled';
+import { verifyApiAccess } from '../lib/license_api_access';
+
+const rulesClient = rulesClientMock.create();
+
+jest.mock('../lib/license_api_access', () => ({
+  verifyApiAccess: jest.fn(),
+}));
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('bulkDeleteRulesRoute', () => {
+  const bulkDeleteRequest = { filter: '' };
+  const bulkDeleteResult = { errors: [], total: 1 };
+
+  it('should delete rules with proper parameters', async () => {
+    const licenseState = licenseStateMock.create();
+    const router = httpServiceMock.createRouter();
+
+    bulkDeleteRulesRoute({ router, licenseState });
+
+    const [config, handler] = router.patch.mock.calls[0];
+
+    expect(config.path).toBe('/internal/alerting/rules/_bulk_delete');
+
+    rulesClient.bulkDeleteRules.mockResolvedValueOnce(bulkDeleteResult);
+
+    const [context, req, res] = mockHandlerArguments(
+      { rulesClient },
+      {
+        body: bulkDeleteRequest,
+      },
+      ['ok']
+    );
+
+    expect(await handler(context, req, res)).toEqual({
+      body: {
+        total: 1,
+        errors: [],
+      },
+    });
+
+    expect(rulesClient.bulkDeleteRules).toHaveBeenCalledTimes(1);
+    expect(rulesClient.bulkDeleteRules.mock.calls[0]).toEqual([bulkDeleteRequest]);
+
+    expect(res.ok).toHaveBeenCalled();
+  });
+
+  it('ensures the license allows bulk deleting rules', async () => {
+    const licenseState = licenseStateMock.create();
+    const router = httpServiceMock.createRouter();
+
+    rulesClient.bulkDeleteRules.mockResolvedValueOnce(bulkDeleteResult);
+
+    bulkDeleteRulesRoute({ router, licenseState });
+
+    const [, handler] = router.patch.mock.calls[0];
+
+    const [context, req, res] = mockHandlerArguments(
+      { rulesClient },
+      {
+        body: bulkDeleteRequest,
+      }
+    );
+
+    await handler(context, req, res);
+
+    expect(verifyApiAccess).toHaveBeenCalledWith(licenseState);
+  });
+
+  it('ensures the license check prevents bulk deleting rules', async () => {
+    const licenseState = licenseStateMock.create();
+    const router = httpServiceMock.createRouter();
+
+    (verifyApiAccess as jest.Mock).mockImplementation(() => {
+      throw new Error('Failure');
+    });
+
+    bulkDeleteRulesRoute({ router, licenseState });
+
+    const [, handler] = router.patch.mock.calls[0];
+
+    const [context, req, res] = mockHandlerArguments(
+      { rulesClient },
+      {
+        body: bulkDeleteRequest,
+      }
+    );
+
+    expect(handler(context, req, res)).rejects.toMatchInlineSnapshot(`[Error: Failure]`);
+  });
+
+  it('ensures the rule type gets validated for the license', async () => {
```

Comment:
I wrote the new endpoint like this because  the old delete endpoint does not have it.  But now I see that most of endpoints catch `RuleTypeDisabledError`, so I added it:  0e7beab5c6c88b8feed22672175448f2ea011dd6

Reply:
I think the old delete endpoint doesn't have it because the rules client does not check for `this.ruleTypeRegistry.ensureRuleTypeEnabled` during deletion, which I actually think is ok. Most likely it's in the bulk delete function because it was copied over from bulk edit? I would probably remove the check from the rules client bulk delete and remove that error handling from the route and remove this test. If for some reason, the rule type becomes disabled for license reasons, we shouldn't prevent users from deleting those rules.